### PR TITLE
Depletion in apertures

### PR DESCRIPTION
**Pull request by wmfw23**
to accompany pre-processed fields added in https://gitlab.cosma.dur.ac.uk/EAGLE/swift-colibre/-/merge_requests/167

### CODE REVIEW

File: velociraptor/catalogue/registration.py [hunk 1]
```diff
@@ -900,7 +900,97 @@ def registration_gas_H_and_He_masses(
...
+    field_path: str, unit_system: VelociraptorUnits
+) -> (unyt.Unit, str, str):
+    """
+    Registers the masses in Hydrogen & Helium within apertures
+    """
+
+    unit = unit_system.mass
+
+    # Capture aperture size
+    match_string = "Aperture_([a-zA-Z]*)_aperture_total_gas_([0-9]*)_kpc"
```
Comment (JBorrow) [inline velociraptor/catalogue/registration.py:913]:
```suggestion
    match_string = "Aperture_Diffuse([a-zA-Z]*)MassesFrom(Table|Model)_aperture_total_gas_([0-9]*)_kpc"
```

File: velociraptor/catalogue/registration.py [hunk 2]
```diff
@@ -900,7 +900,97 @@ def registration_gas_H_and_He_masses(
...
+    regex = cached_regex(match_string)
+
+    match = regex.match(field_path)
+
+    if match:
+        long_element = match.group(1)
+        aperture_size = match.group(2)
```
Comment (JBorrow) [inline velociraptor/catalogue/registration.py:920]:
```suggestion
        element = match.group(1)
        table_model = match.group(2)
        aperture_size = match.group(3)
```

File: velociraptor/catalogue/registration.py [hunk 3]
```diff
@@ -900,7 +900,97 @@ def registration_gas_H_and_He_masses(
...
+                         "DiffuseOxygenMassesFromTable": "M^{\\rm table}_{\\rm O,\. diffuse}",
+                         "DiffuseMagnesiumMassesFromTable": "M^{\\rm table}_{\\rm Mg,\. diffuse}",
+                         "DiffuseSiliconMassesFromTable": "M^{\\rm table}_{\\rm Si,\. diffuse}",
+                         "DiffuseIronMassesFromTable": "M^{\\rm table}_{\\rm Fe,\. diffuse}",
+                         "DiffuseCarbonMassesFromModel": "M^{\\rm model}_{\\rm C,\. diffuse}",
+                         "DiffuseOxygenMassesFromModel": "M^{\\rm model}_{\\rm O,\. diffuse}",
+                         "DiffuseMagnesiumMassesFromModel": "M^{\\rm model}_{\\rm Mg,\. diffuse}",
+                         "DiffuseSiliconMassesFromModel": "M^{\\rm model}_{\\rm Si,\. diffuse}",
+                         "DiffuseIronMassesFromModel": "M^{\\rm model}_{\\rm Fe,\. diffuse}",
+                         }[long_element]
```
Comment (JBorrow) [inline velociraptor/catalogue/registration.py:958]:
```suggestion
            long_name_element = f"Diffuse {element} from depletion {table_model.lower()}"
            element_symbol = {
              "Carbon": "C", "Oxygen": "O", ...
            }[element]
            math_name = f"M^{{\\rm {table_model.lower()}}}_{{\\rm {element_symbol}, diffuse}}"
```

File: velociraptor/catalogue/registration.py [hunk 4]
```diff
@@ -900,7 +900,97 @@ def registration_gas_H_and_He_masses(
...
+        except KeyError:
+            raise RegistrationDoesNotMatchError
+
+        full_name = f"{long_name_element} Gas Mass {math_name} ({aperture_size} kpc)"
+        snake_case = f"{short_element}_mass_{aperture_size}_kpc"
```
Comment (JBorrow) [inline velociraptor/catalogue/registration.py:963]:
```suggestion
        full_name = f"{element} ({table_model}) Gas Mass {element_symbol} ({aperture_size} kpc)"
        snake_case = f"{element.lower()}_mass_{table_model.lower()}_{aperture_size}_kpc"
```

File: velociraptor/catalogue/registration.py [hunk 5]
```diff
@@ -900,7 +900,97 @@ def registration_gas_H_and_He_masses(
...
+
+    unit = unit_system.mass
+
+    match_string = "Aperture_DustMassesFromTable_aperture_total_gas_([0-9]*)_kpc"
+    regex = cached_regex(match_string)
+    match = regex.match(field_path)
+
+    if match:
+        aperture_size = match.group(1)
+        full_name = f"Total Dust Mass from tables ({aperture_size} kpc)"
```
Comment (JBorrow) [inline velociraptor/catalogue/registration.py:987]:
```suggestion
        full_name = f"Total Dust Mass from Tables ({aperture_size} kpc)"
```

**Review by JBorrow — COMMENTED** [review]
We can save a lot of hardcoding here with a little regex.

**Comment by wmfw23** [comment]
well I was following what's done for `colibre/scripts/load_sfh_data.py`

**Comment by wmfw23** [comment]
Thanks, I also rebased to Evgenii's upcoming PR to make merging easier

**Comment by JBorrow** [comment]
Happy with this now once we're done with Evgenii's stuff. If @EvgeniiChaikin is happy we can actually just merge this into his branch.

**Comment by EvgeniiChaikin** [comment]
Looks to me like something is wrong with the versions. @wmfw23 you will probably need to try to rebase again.

**Comment by JBorrow** [comment]
No, it's based on your branch, so there should be merge conflicts. We can merge this into `rearrange_colibre_registration` without problems

**Comment by JBorrow** [comment]
Actually, I'm wrong - @wmfw23 what branch did you rebase to?

**Comment by wmfw23** [comment]
the open `fix_typo_in_H2_HI_mass_reg`, guess I just need to rebase to master instead? no other open PRs?

**Comment by wmfw23** [comment]
@EvgeniiChaikin does it work for you now?

**Comment by EvgeniiChaikin** [comment]
@wmfw23 All looks good to me! 

I just double-checked that everything works, see http://swift.dur.ac.uk/COLIBREPlots/chaikin/individual_runs/NEWEST/L006N188/NO_EOS/z0.0_hawk_AGN_dT_3e8_L006N188_h2_hi_version2/

**Pull request merged**